File: backend/src/models/Token.ts
import mongoose, { Schema, InferSchemaType } from 'mongoose';

const TokenSchema = new Schema({
  provider: { type: String, required: true, index: true },
  refreshToken: { type: String, required: true },
  accessToken: { type: String },
  scope: { type: String },
  tokenType: { type: String },
  expiryDate: { type: Number },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now },
});

export type TokenDoc = InferSchemaType<typeof TokenSchema> & { _id: string };

export default mongoose.model('Token', TokenSchema);

File: backend_py/app/models/token.py
from datetime import datetime, timezone
from typing import Optional

from beanie import Document
from pydantic import Field


def _utcnow() -> datetime:
    return datetime.now(timezone.utc)


class Token(Document):
    """Mirrors the Mongoose `Token` model (see note in job.py about casing)."""

    provider: str
    refreshToken: str
    accessToken: Optional[str] = None
    scope: Optional[str] = None
    tokenType: Optional[str] = None
    expiryDate: Optional[int] = None
    createdAt: datetime = Field(default_factory=_utcnow)
    updatedAt: datetime = Field(default_factory=_utcnow)

    class Settings:
        name = "tokens"
        indexes = ["provider"]
